lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Implements the plugin manager class."""
+
+from typing import Dict, List
+
+from charms.opensearch.v0.opensearch_plugins import OpenSearchPlugin
+from ops.framework import Object
+from ops.model import ActiveStatus, StatusBase
+
+# The unique Charmhub library identifier, never change it
+LIBID = "da838485175f47dbbbb83d76c07cab4c"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+OpenSearchPluginsAvailable = {}
```

Comment:
I like the idea of having a registry! 
The `Available` part may be a bit confusion (built-in, configured etc..). we could perhaps rename it to: `ConfigExposedPlugins`? or something else that's explicit